src/main/java/past/cwdt/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -1,29 +1,70 @@
 package past.cwdt.bookstore.service.impl;
 
 import java.util.List;
-import org.springframework.beans.factory.annotation.Autowired;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
+import past.cwdt.bookstore.dto.BookDto;
+import past.cwdt.bookstore.dto.BookDtoSearchByParameters;
+import past.cwdt.bookstore.dto.CreateBookRequestDto;
+import past.cwdt.bookstore.exceptions.EntityNotFoundException;
+import past.cwdt.bookstore.mapper.BookMapper;
 import past.cwdt.bookstore.model.Book;
 import past.cwdt.bookstore.repository.BookRepository;
+import past.cwdt.bookstore.repository.impl.BookSpecificationBuilder;
 import past.cwdt.bookstore.service.BookService;
 
+@RequiredArgsConstructor
 @Service
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+    private final BookSpecificationBuilder bookSpecificationBuilder;
 
-    @Autowired
-    public BookServiceImpl(BookRepository bookRepository) {
-        this.bookRepository = bookRepository;
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        return bookMapper.toDto(bookRepository.save(bookMapper.toBook(requestDto)));
+    }
+
+    @Override
+    public BookDto getBookById(Long id) {
+        return bookRepository.findById(id)
+                .map(bookMapper::toDto)
+                .orElseThrow(()
+                        -> new EntityNotFoundException("Can't find book by Id " + id));
     }
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
 
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+        checkIfBookExistsById(id);
```

Comment:
I would suggest to find book by Id, then by using Mapper mapped requestDto to book

Book bookToUpdate = bookRepository.findById(id)
                .orElseThrow(() -> new NoSuchElementException("Can't find book by id: " + id));
        bookMapper.updateBook(requestDto, bookToUpdate);
        return bookMapper.toDto(bookRepository.save(bookToUpdate));